### DISCUSSION TITLE
Definition of "Tests" in GitLab CI

### DISCUSSION THREAD

**Discussion by DBS-ST-VIT** [renovatebot/renovate#25402] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.31.5

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I actually want to know, what kind of "Tests" renovate is looking for and dont taking care anymore, if i set ignoreTests to true.

### What you tried so far.

I am currently facing some Merge Requests, that should be auto-merged in theory, but aint auto merged.

In the docs i read, that there is a parameter called ignoreTests. All projects, where the problematic merge requests occur, are not having any tests.

So i am very sure, that this is the reason for the not working automerge.

But what i am very unsure about, is, how renovate defines "Tests" in context of GitLab CI? Does it check a stage called "test" and the jobs in this stage, or what is it actually doing? I am basically searching for the things, that renovate ignores, if i pass ignoreTests = true.

**Comment by viceice** [comment]
renovate won't wait for gitlab pipeline to succeed and will merge right away when ignore tests is used. you should add some minimal gitlab pipeline to generate status checks which will be read / used by renovate

**Reply by DBS-ST-VIT** [reply]
I understood that. And i dont want to turn ignoreTests on. My question was, whats the specific indicator, that renovate is checking. What means status checks?

Is it only a check, whether the pipeline succeeded? Or does it specifically check, whether a stage with a specific name succeeded? What is renovate doing there?

**Reply by rarkins** [reply]
It looks for succeeded status checks on its branch commit

**Reply by DBS-ST-VIT** [reply]
But what is the definition of status checks? Does it mean, the full pipeline ran successfully?

**Reply by rarkins** [reply, marked as answer]
https://github.com/renovatebot/renovate/blob/bf0ba5e8319b4a5e57e0f0d7ca38201698308210/lib/modules/platform/gitlab/index.ts#L411-L435

**Reply by DBS-ST-VIT** [reply]
Thanks for your response. Thats what i searched for.
I also found this part of the GitLab docs, which is also a good information source: https://docs.gitlab.com/ee/api/commits.html#commit-status
